### ISSUE TITLE
Upgrading from commercetools 1.5.1 to 1.6.9 causes an error for destination.region

### ISSUE THREAD

**Issue by chawii** [labd/terraform-provider-commercetools#366]
**Information**
Terraform version: 1.2.8
Provider version: 1.5.1 / 1.6.9
Resource: commercetools_subscription

**To Reproduce**
1. Start with commercetools provider version 1.5.1
2. Change commercetools provider version to 1.6.9
3. terraform init -upgrade
4. terraform plan
5. terraform apply

```hcl
# main.tf
terraform {
  required_providers {
    commercetools = {
      source  = "labd/commercetools"
      version = "1.6.9"
    }
  }
}

# subscription.tf
resource "commercetools_subscription" "sample_subscription" {
  key = "sample_subscription"

  destination {
    type       = "GoogleCloudPubSub"
    project_id = "project_id_here"
    topic      = "topic_name_here"
    region     = "us-east1"
  }

  format {
    type                 = "CloudEvents"
    cloud_events_version = "1.0"
  }

  changes {
    resource_type_ids = ["product"]
  }

  message {
    resource_type_id = "product"
    types = [
      "ProductCreated",
      "ProductPublished"
    ]
  }
}
```

Terraform plan results after updating to 1.6.9:
```
Terraform will perform the following actions:

  # commercetools_subscription.sample_subscription will be updated in-place
  ~ resource "commercetools_subscription" "sample_subscription" {
      ~ id      = "59a1aea3-e6a3-4c62-8a22-e09e6dde2023" -> (known after apply)
      ~ version = 2 -> (known after apply)
        # (1 unchanged attribute hidden)

      ~ destination {
          + region     = "us-east1" 
            # (3 unchanged attributes hidden)
        }

        # (16 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Executing terraform apply throws this error below:
```
commercetools_subscription.sample_subscription: Modifying... [id=59a1aea3-e6a3-4c62-8a22-e09e6dde2023]
╷
│ Error: Provider produced inconsistent result after apply
│
│ When applying changes to commercetools_subscription.sample_subscription, provider "provider[\"registry.terraform.io/labd/commercetools\"]"
│ produced an unexpected new value: .destination[0].region: was cty.StringVal("us-east1"), but now null.
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by philippeckelintive** [comment]
@chawii commercetools API does not support `region` in `destination` if using `GoogleCloudPubSub`; see here: https://docs.commercetools.com/api/projects/subscriptions#googlecloudpubsubdestination

I assume the provider - wrongly - did accept this attribute `region` and never used it for `GoogleCloudPubSub` anyways.

TLDR: remove `region     = "us-east1"` and you should be fine.

**Comment by chawii** [comment]
> @chawii commercetools API does not support `region` in `destination` if using `GoogleCloudPubSub`; see here: https://docs.commercetools.com/api/projects/subscriptions#googlecloudpubsubdestination
> 
> I assume the provider - wrongly - did accept this attribute `region` and never used it for `GoogleCloudPubSub` anyways.
> 
> TLDR: remove `region = "us-east1"` and you should be fine.

Hi @philippeckelintive thank you for your reply I tried without the region and it worked! 

Thank you, I'll close this ticket as completed.
